src/test/java/com/aws/greengrass/authorization/WildcardTrieTest.java

```diff
@@ -16,51 +16,111 @@
 class WildcardTrieTest {
     @Test
     void testGlobWildcardMatching() {
+        // no wildcards
         WildcardTrie rt = new WildcardTrie();
-        rt.add("abc*xyz*");
-        assertTrue(rt.matchesMQTT("abc1234asdxyz456" ));
-        assertTrue(rt.matchesMQTT("abcdxyz"));
-        assertFalse(rt.matchesMQTT("abc123xyz456/89"));
+        rt.add("nowildcard");
+        assertTrue(rt.matchesStandard("nowildcard"));
+        assertFalse(rt.matchesStandard("topic"));
+
+        assertTrue(rt.matchesMQTT("nowildcard"));
+        assertFalse(rt.matchesMQTT("topic"));
+
+        // Test wildcards in middle
+        rt.add("abc*xy*z");
+        assertTrue(rt.matchesStandard("abc123xyabc!@/<>#$%^&*()_+-=z" ));
+        assertTrue(rt.matchesStandard("abcxyz"));
+        assertTrue(rt.matchesStandard("abcxy/ 7895z"));
+        assertTrue(rt.matchesStandard("abc123xy90zABCz"));
+        assertTrue(rt.matchesStandard("abc123xy90zABCxyABCz"));
```

Comment:
Good test cases. Also add one that has a partial match after the first glob. E.g. `abc123x123xyz`